src/plugins/embeddable/public/lib/containers/embeddable_child_panel.tsx

```diff
@@ -41,19 +51,64 @@ export class EmbeddableChildPanel extends React.Component<EmbeddableChildPanelPr
   constructor(props: EmbeddableChildPanelProps) {
     super(props);
     this.state = {
-      loading: true,
+      firstTimeLoading: true,
     };
 
     this.mounted = false;
   }
 
+  private getEventStatus(output: EmbeddableOutput): EmbeddableRenderStatus {
+    if (output.rendered === true) {
+      return 'rendered';
+    } else if (output.loading === false) {
+      return 'loaded';
+    } else if (output.error !== undefined) {
```

Comment:
Based on your description of the state machine in the PR description, `error` also indicates any errors during rendering. In that case, according to my understanding, the flags would be:

* `rendered === false`
* `loading === false`
* `error !== undefined`

However, this method returns "loaded" in this case. If we take a peek at `onContainerError`, `rendered` is actually set to `true` so the state that I'm describing cannot happen. I still think we would make this code more robust by first checking for errors, i.e.:

```typescript
if (output.error !== undefined) {
    return 'error';
else if (output.rendered === true) {
    return 'rendered';
...
```

In any case, I think we need test cases for this.